### PR TITLE
[WEB] [API] Application back

### PR DESCRIPTION
**Pull request by karen-baltazar**
- [x] Create Application_Documents queries.
- [x] Create Application_Documents mutations.
- [x] Updated Application queries and mutations.
- [x] Back-end and Front-end connection for Application Documentation Review.

### CODE REVIEW

File: api/source/graphql/mutations/ApplicationDocumentsMutations.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        });
+
+      const newApplicationDocument = await db.select().from(APPLICATION_DOCUMENTS_TABLE_NAME).where("id", id);
+
+      return {
+        ...newApplicationDocument[0],
+        application: myApplication[0],
+        document: myDocument[0],
+      };
+    },
```
Comment (iangg29) [inline api/source/graphql/mutations/ApplicationDocumentsMutations.ts:62]: Unnecessary mutation, `pivot` table should only be used to link models. This `link` part should happen automatically on `Document` or `Application` creation.  Also, within `Document` and `Application` queries opposite objects must be fetched using `pivot` table.

File: api/source/graphql/mutations/ApplicationMutations.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      });
+      console.log("NO HAY ERROR 3 ---------------------------------------")
+      
+      const myCitation = await db.select().from(CITATION_TABLE_NAME).where({ id: citation_id })
+      .catch((error: Error) => {
+        console.error(error);
+        throw new GraphQLError(error.name);
+      });
+      console.log("NO HAY ERROR 4 ---------------------------------------")
+
```
Comment (iangg29) [inline api/source/graphql/mutations/ApplicationMutations.ts:92]: Remove debugging `console.log`.

File: api/source/graphql/mutations/AuthMutations.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      } catch (error: any) {
+        return {
+          success: false,
+          error: error.message,
+          user: null,
+        };
+      }
+    },
+  },
+};
```
Comment (iangg29) [inline api/source/graphql/mutations/AuthMutations.ts:135]: Authentication is no longer handled with graphql. Remove file.

File: api/source/graphql/mutations/UserMutations.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        })
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+      const newUser = await db.select().from(USER_TABLE_NAME).where({ id });
+      return newUser[0];
+    },
+  },
+};
```
Comment (iangg29) [inline api/source/graphql/mutations/UserMutations.ts:50]: Authentication is no longer handled with graphql. Remove file.

File: api/source/graphql/queries/ApplicationQueries.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  CITATION_TABLE_NAME,
+} from "../../database/utils/database_constants";
+
+export default {
+  applications: {
+    type: GraphQLList(ApplicationType),
+    resolve: () => {
+      return db.select().table(APPLICATION_TABLE_NAME);
+    },
+  },
```
Comment (iangg29) [inline api/source/graphql/queries/ApplicationQueries.ts:19]: Pull relationship objects.

File: api/source/types/ApplicationLabelType.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    // },
+    // created_at: {
+    //   type: GraphQLNonNull(GraphQLString),
+    //   description: "Date in which label was assigned to the application",
+    // },
+    // updated_at: {
+    //   type: GraphQLNonNull(GraphQLString),
+    //   description: "Date in which label was last updated",
+    // }
+  },
```
Comment (iangg29) [inline api/source/types/ApplicationLabelType.ts:30]: Remove unused code. This Type was already assigned to @alfredohuerta

File: web/package.json [hunk 7]
```diff
@@ -21,6 +21,7 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-icons": "^4.4.0",
+    "react-modal": "^3.15.1",
     "react-redux": "^8.0.2",
     "react-relay": "^14.1.0",
     "react-router-dom": "^6.3.0",
```
Comment (iangg29) [inline web/package.json:27]: Modals are being implemented with [flowbite](https://flowbite-react.com/modal).

File: web/src/components/RequestCard/RequestCard.tsx [hunk 8]
```diff
@@ -43,7 +57,7 @@ const RequestCard = ({ image, proyectTile, announcement, user, label, buttonText
                 className="rounded-lg px-5 py-2.5 text-center text-sm font-medium text-white hover:scale-110"
                 style={{ backgroundColor: color }}
               >
-                <Link to={"/app/Solicitudes/RevisarDocumentos/Documentos"}>{buttonText}</Link>
+                <Link to={"/app/applications/reviewdocuments/documents"}>{buttonText}</Link>
               </button>
             </div>
```
Comment (iangg29) [inline web/src/components/RequestCard/RequestCard.tsx:62]: Update route names. (English & lowercase).

File: web/src/components/includes/navBar/ArrowDrop.tsx [hunk 9]
```diff
@@ -11,23 +11,29 @@ const ArrowDrop = (): JSX.Element => {
...
-              <Link to={""} className="navBarLink">
+              <Link to={"applications/reviewapproved"} className="navBarLink">
                 Aprobadas
               </Link>
             </Dropdown.Item>
 
+            <Dropdown.Item>
+              <Link to={"applications/reviewfinished"} className="navBarLink">
+                Finalizadas
+              </Link>
```
Comment (iangg29) [inline web/src/components/includes/navBar/ArrowDrop.tsx:34]: Modify route names (English & lowercase).

File: web/src/pages/Req_Detail.tsx [hunk 10]
```diff
@@ -4,49 +4,62 @@ import Document from "../components/Doc_Review";
...
+        }
+        applicationdocuments(application_id: $application_id) {
+          file_name
+          url
+          updated_at
+        }
+      }
+    `,
+    { application_id: "8bc874cc-e021-4504-ac90-a473e57e999c" },
+  );
```
Comment (iangg29) [inline web/src/pages/Req_Detail.tsx:37]: Add document list object within application model and handle fetching inside the query.

File: web/src/pages/suppRequests/ApplicationAccepted.tsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+        {empty ? (
+          <h1 className="col-span-3 text-center">
+            <br />
+            No hay solicitudes aprobadas.
+          </h1>
+        ) : (
+          applicationByStatusID?.map((element: any) => (
+            <RequestCard
+              key={element.id}
+              image={back}
```
Comment (iangg29) [inline web/src/pages/suppRequests/ApplicationAccepted.tsx:58]: Split conditional rendering condition.

File: web/src/pages/suppRequests/ApplicationFinalized.tsx [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+          citation {
+            id
+            title
+          }
+        }
+      }
+    `,
+    { application_status_id: "" },
+  );
+
```
Comment (iangg29) [inline web/src/pages/suppRequests/ApplicationFinalized.tsx:35]: If query arguments can be sent null modify schema and handle null values inside query resolver.

File: web/src/pages/suppRequests/ApplicationFinalized.tsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+          <h1 className="col-span-3 text-center">
+            <br />
+            No hay solicitudes finalizadas.
+          </h1>
+        ) : (
+          applicationByStatusID?.map((element: any) => (
+            <RequestCard
+              key={element.id}
+              image={back}
+              proyectTile={element.title}
```
Comment (iangg29) [inline web/src/pages/suppRequests/ApplicationFinalized.tsx:59]: Simplify conditional rendering condition.

File: web/src/pages/suppRequests/ApproveApplication.tsx [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+            id
+            title
+          }
+        }
+      }
+    `,
+    { application_status_id: "" },
+  );
+
+  const exampleLabels = [{ label: "Cultura" }, { label: "Baile" }];
```
Comment (iangg29) [inline web/src/pages/suppRequests/ApproveApplication.tsx:33]: If query arguments can be sent null modify schema and handle null values inside query resolver.

File: web/src/pages/suppRequests/ApproveApplication.tsx [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+        {empty ? (
+          <h1 className="col-span-3 text-center">
+            <br />
+            No hay solicitudes pendientes de revisión de propuesta.
+          </h1>
+        ) : (
+          applicationByStatusID?.map((element: any) => (
+            <RequestCard
+              key={element.id}
+              image={back}
```
Comment (iangg29) [inline web/src/pages/suppRequests/ApproveApplication.tsx:55]: Simplify conditional rendering condition.

File: web/src/pages/suppRequests/ApproveDoc.tsx [hunk 16]
```diff
@@ -1,42 +1,69 @@
...
+          citation {
+            id
+            title
+          }
+        }
+      }
+    `,
+    { application_status_id: "58a765a1-1204-48a5-b81c-c8c711aceb18" },
+  );
+
```
Comment (iangg29) [inline web/src/pages/suppRequests/ApproveDoc.tsx:32]: This change will **break produciton**. Fixed IDs should **never** be implemented.

**Review by iangg29 — CHANGES_REQUESTED** [review]

File: api/source/graphql/mutations/ApplicationMutations.ts [hunk 17]
```diff
@@ -0,0 +1,247 @@
...
+    type: GraphQLBoolean,
+    args: {
+      id: {
+        type: GraphQLNonNull(GraphQLID),
+      },
+    },
+    resolve: async (_: any, { id }: any) => {
+      await db(APPLICATION_TABLE_NAME).where("id", id).del();
+      return true;
+    },
```
Comment (alfredohuerta) [inline api/source/graphql/mutations/ApplicationMutations.ts:245]: Before deleting the application, check if the application exist

File: api/source/graphql/queries/UserQueries.ts [hunk 18]
```diff
@@ -12,4 +12,20 @@ export default {
...
+      },
+    },
+    resolve: async (_: any, { id }: any) => {
+      const user = await db.select().from(USER_TABLE_NAME).where({ id })
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+      return user[0];
+    },
```
Comment (alfredohuerta) [inline api/source/graphql/queries/UserQueries.ts:29]: Delete un-used query

File: api/source/types/ApplicationLabelType.ts [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+  name: "ApplicationLabel",
+  description:
+    "Application - Label model, contains all the information related to the labels assigned to an application",
+  fields: {
+    id: {
+      type: GraphQLNonNull(GraphQLID),
+      description: "ApplicationLabel ID",
+    },
+  },
+});
```
Comment (alfredohuerta) [inline api/source/types/ApplicationLabelType.ts:15]: Delete unused type

File: api/source/types/ApplicationType.ts [hunk 20]
```diff
@@ -50,6 +63,10 @@ export const ApplicationType = new GraphQLObjectType({
       type: GraphQLNonNull(CitationType),
       description: "Citation of the application",
     },
+    applicationDocuments: {
+      type: GraphQLList(DocumentType),
+      description: "Application documents",
+    },
```
Comment (alfredohuerta) [inline api/source/types/ApplicationType.ts:69]: Add labeltype as a GraphQLList

File: web/src/pages/Chat/Chat.tsx [hunk 21]
```diff
@@ -0,0 +1,7 @@
+// (c) Tecnologico de Monterrey 2022, rights reserved.
+
+const Chat = (): JSX.Element => {
+  return <div></div>;
+};
+
+export default Chat;
```
Comment (alfredohuerta) [inline web/src/pages/Chat/Chat.tsx:7]: Delete unused view

File: web/src/routes/index.ts [hunk 22]
```diff
@@ -17,23 +22,43 @@ const routes: iRoute[] = [
...
   },
   {
-    path: "/Solicitudes/RevisarDocumentos",
+    path: "/analytics",
+    components: Analytic,
+  },
+  {
+    path: "/chat",
+    components: Chat,
+  },
```
Comment (alfredohuerta) [inline web/src/routes/index.ts:35]: delete unused path

**Review by alfredohuerta — CHANGES_REQUESTED** [review]
Change comments

File: api/source/graphql/queries/UserQueries.ts [hunk 23]
```diff
@@ -1,15 +1,25 @@
...
     },
-  },
+    resolve: async (_: any, { id }: any) => {
+      const user = await db.select().from(USER_TABLE_NAME).where({ id })
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+      return user[0];
+    },
```
Comment (alfredohuerta) [inline api/source/graphql/queries/UserQueries.ts:23]: delete unused query

**Review by alfredohuerta — CHANGES_REQUESTED** [review]

**Review by alfredohuerta — APPROVED** [review]

**Pull request merged**